src/sync.d

```diff
@@ -2155,40 +2157,49 @@ final class SyncEngine
 						SysTime localModifiedTime = timeLastModified(oldPath).toUTC();
 						localModifiedTime.fracSecs = Duration.zero;
 						item.mtime.fracSecs = Duration.zero;
-						
+
+						log.vdebug("Local modified time: ", localModifiedTime);
+						log.vdebug("Item modified time:  ", item.mtime);
+
 						// Compare file on disk modified time with modified time provided by OneDrive API
-						if (localModifiedTime > item.mtime) {
+						if (localModifiedTime >= item.mtime) { 			// changed > to >=
 							// local file is newer than item on OneDrive
 							// no local rename
 							// no download needed
+							log.vlog("Local time >= item time : local file should be newer to remote.");
+							if (localModifiedTime == item.mtime) {
+								log.vlog("Local time is equal to item modified time: the file was probably modified by another process on the local disk.");
+							}
 							log.vlog("Local item modified time is newer based on UTC time conversion - keeping local item");
 							log.vdebug("Skipping OneDrive change as this is determined to be unwanted due to local item modified time being newer than OneDrive item");
 							skippedItems ~= item.id;
 							return;
 						} else {
 							// remote file is newer than local item
 							log.vlog("Remote item modified time is newer based on UTC time conversion"); // correct message, remote item is newer
-							log.vdebug("localModifiedTime (local file): ", localModifiedTime);
-							log.vdebug("item.mtime (OneDrive item):     ", item.mtime);
-							
+							//log.vdebug("localModifiedTime (local file): ", localModifiedTime);
+							//log.vdebug("item.mtime (OneDrive item):     ", item.mtime);
+							log.vlog("local file modified time: ", localModifiedTime);
+							log.vlog("OneDrive database item time (item.mtime): ", item.mtime);
```

Comment:
item.mtime is not the database item - but the item provided by OneDrive via the API